fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/client/item/v1/FabricItemUpdateAnimationHandlers.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.item.v1;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import net.minecraft.item.Item;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+
+@Environment(EnvType.CLIENT)
+public class FabricItemUpdateAnimationHandlers {
+	private static final Map<Item, UpdateAnimationHandler> handlers = new HashMap<>();
+	private static final UpdateAnimationHandler DEFAULT = (original, updated) -> true;
+
+	public static void register(Item item, UpdateAnimationHandler handler) {
+		if (handlers.containsKey(item)) {
+			Identifier registryID = Registry.ITEM.getId(item);
+			throw new UnsupportedOperationException(String.format("Attempted to register an Item Update Animation Handler for %s, but one was already registered!", registryID.toString()));
```

Comment:
maybe allow intentional overriding? do you think it's worth it?

Reply:
This was something I considered but wasn't sure about. I leaned towards having it throw an error so you _know_ your mod might be conflicting with another one (instead of a silent fail)... otherwise it's basically just a race to see which mod loads first. I don't have a major preference on this.

Reply:
The stuff related to juuz's PR to allow use of minecraft's built-in item renderer model (such as for block entities) I think does not allowed duplicate entries right now

Reply:
I think it's less about duplicate entries and more about throwing an error when more than 1 people try to register a handler vs. letting the second caller override the first (unless I misunderstood liach's comment). Allowing multiple handlers, eg. `List<UpdateAnimationHandler>` and checking each, basically just puts us in event territory (with less overhead), so I don't think it would make much sense.

Reply:
Yeah a list won't work out here.